packages/main/src/ToolbarItem.ts

```diff
@@ -0,0 +1,66 @@
+import UI5Element from "@ui5/webcomponents-base/dist/UI5Element.js";
+import { TemplateFunction } from "@ui5/webcomponents-base/dist/renderer/executeTemplate.js";
+import property from "@ui5/webcomponents-base/dist/decorators/property.js";
+import customElement from "@ui5/webcomponents-base/dist/decorators/customElement.js";
+
+import ToolbarPriority from "./types/ToolbarPriority.js";
+
+@customElement({
+	tag: "ui5-tb-item",
+})
+
+/**
+ * @class
+ * The <code>ui5-tb-item</code> represents an abstract class for items,
+ * used in the <code>ui5-toolbar</code>.
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias ToolbarItem
+ * @extends UI5Element
+ * @public
+ */
+class ToolbarItem extends UI5Element {
+	/**
+	 * Property used to define the access of the item to the overflow Popover. If "Never" option is set,
+	 * the item never goes in the Popover, if "Always" - it never comes out of it.
+	 * @public
+	 * @defaultvalue ToolbarPriority.Default,
+	 */
+	@property({ type: ToolbarPriority })
+	priority!: string;
+
+	/**
+	 * When set, the button will not be visible in the toolbar
+	 * @private
+	 */
+	@property({ type: Boolean })
+	hidden!: boolean;
```

Comment:
`hidden` is an existing DOM API. It's best not to mess up with these. I would suggest:
 - either to make this `_hidden`
 - or to use the native `hidden`, but don't declare a property here, as it already exists on HTMLElement I think (or is it only for some HTML elements)